Auth0/CredentialsManager.swift

```diff
@@ -53,11 +53,31 @@ public struct CredentialsManager {
     /// Store credentials instance in keychain
     ///
     /// - Parameter credentials: credentials instance to store
-    /// - Returns: Bool outcome of success
+    /// - Returns: if credentials were stored
     public func store(credentials: Credentials) -> Bool {
         return self.storage.setData(NSKeyedArchiver.archivedData(withRootObject: credentials), forKey: storeKey)
     }
 
+    /// Clear credentials stored in keychain
+    ///
+    /// - Returns: if credentials were removed
+    public func clear() -> Bool {
+        return self.storage.deleteEntry(forKey: storeKey)
+    }
+
+    /// Checks if valid credentials are stored
+    ///
+    /// - Returns: if there are valid and non-expired credentials stored
+    public func hasValid() -> Bool {
+        guard
+            let data = self.storage.data(forKey:self.storeKey),
+            let credentials = NSKeyedUnarchiver.unarchiveObject(with: data) as? Credentials,
+            credentials.accessToken != nil,
+            let expiresIn = credentials.expiresIn
+            else { return false }
+        return expiresIn > Date()
```

Comment:
This is one of the conditions. But if `expiresIn <= Date() && credentials.refreshToken != nil` you will be able to refresh it, so that's another case that returns "valid credentials". No network call required here.